### PR TITLE
Remember home page section state

### LINKED ISSUE #3200: Home page trending section: Remember state when going back
It's a bit annoying when you want to check out multiple apps in e.g. the "New" section that this part of the page resets when you go back. This happens on both desktop with the tabs, and mobile with the dropdown.

- Open flathub.org
- Click one of the non-default tabs in the new trending/popular/new/updated section, e.g. "New"
- Open one of the apps in that section
- Go back in the browser
- You're back on "Trending"

### PR DESCRIPTION
**Pull request by YonatanPerkel**
Closes #3200

### CODE REVIEW

File: frontend/pages/index.tsx [hunk 1]
```diff
@@ -93,12 +94,27 @@ const TopSection = ({
 }) => {
   const { t } = useTranslation()
 
-  const [selectedName, setSelectedName] = useState(topApps[0].name)
+  const [selectedName, setSelectedName] = useState("")
+  const [selectedIndex, setSelectedIndex] = useLocalStorage(
+    "selected-category",
```
Comment (razzeee) [inline frontend/pages/index.tsx:99]: Should probably mention, that it only affects the landing page

File: frontend/pages/index.tsx [hunk 2]
```diff
@@ -93,12 +94,27 @@ const TopSection = ({
...
+    0,
```
Comment (razzeee) [inline frontend/pages/index.tsx:100]: I would prefer to save the name not the index

Comment (YonatanPerkel) [reply frontend/pages/index.tsx]: Changed local storage to store name instead of index.

File: frontend/pages/index.tsx [hunk 3]
```diff
@@ -109,13 +131,15 @@ const TopSection = ({
       appSelection={
         <>
           <MultiToggle
-            items={topApps.map((x) => ({
+            items={topApps.map((x, index) => ({
```
Comment (razzeee) [inline frontend/pages/index.tsx:134]: index can be removed

**Comment by razzeee** [comment]
Thank you, I might need a few days, before I can get back to this and give it it's final testing

**Comment by razzeee** [comment]
This seems to work great.

But I'm unsure, if we really want it like this, as this keeps the state, even if I leave the page and come back days later.

I would much more prefer to always default to trending, if somebody goes directly to flathub.org, but only remember these, when navigating back. I think the "correct" approach for this, would be to place the currently selected item into the url. Similar to how we do it here https://github.com/flathub-infra/website/blob/main/frontend/pages/admin/quality-moderation/index.tsx#L246

**Comment by YonatanPerkel** [comment]
Instead of using url query params, I tried doing it with location hash and I think it's more appropriate for this use case.

**Comment by razzeee** [comment]
> Instead of using url query params, I tried doing it with location hash and I think it's more appropriate for this use case.

I don't feel that way, as it locks us in. There can only be one hash per page - so if we want to have (and remember) a second tab selection on this page, we need to change api and would kill all links, that previously worked.

**Comment by razzeee** [comment]
@YonatanPerkel thank you for looking into this

**Pull request merged**